interpreter/cli/cli.py

```diff
@@ -124,6 +135,10 @@ def cli(interpreter):
         if attr_value is not None and hasattr(interpreter, attr_name):
             setattr(interpreter, attr_name, attr_value)
 
+    # if auto_run is enabled, we won't bother scanning code
+    if interpreter.auto_run and not interpreter.scan_code == "off":
+        setattr(interpreter, "scan_code", "off")
```

Comment:
Not sure if there's a better pattern to handle interconnected attributes like this, but I'm guessing there might be a few things in the future that might depend on the status of `auto_run`, so we might want to explore a less naive approach than what I've done here.